manifests/facts.pp

```diff
@@ -41,10 +41,10 @@
   if $facts["os"]["family"] == "windows" {
     # on windows task scheduler prevent dupes already so no need to handle the PID here
     scheduled_task{"mcollective_facts_yaml_refresh":
-      ensure               => $cron_ensure,
-      command              => $rubypath,
-      arguments            => "'${scriptpath}' -o '${factspath}'",
-      trigger              => {
+      ensure    => $cron_ensure,
+      command   => $rubypath,
+      arguments => "'${scriptpath}' -o '${factspath}' --sleep '${fqdn_rand($refresh_interval, 'facts cronjob')}'",
```

Comment:
Alternatively, I could use the rand_fqdn function in the script template. 
Is that what you would prefer? 

I find it a bit confusing, but I do not have the experience with windows to make the choice.